proposals/theme_manifest.md

```diff
@@ -0,0 +1,60 @@
+# Theme manifest standards proposal
+
+## Description Of Issue
+Theme manifests should be documented and standardised. While themes are not part of Cumcord itself,
+they are an essential part of most users' client mod experience,
+so when theme loader plugins inevitably appear, the manifest (and repo) formats should be standardised and cross-compatible.
+
+## Solution
+Standardise and document the manifest format (new-BD-style), what the fields must be, and what each is for.
+
+## Advantages
+
+ - prevent 3rd parties from adding their own fields to manifests for their own use, cluttering repos.
+ - act as documentation for theme developers on what to include in their manifest
+
+## Disadvantages
+ - locks down possibly helpful info that can be included in manifests
+
+## General Implementation Details
+ - only new BD style manifests should be carried forward into CC themes - not `//META`-style manifests
+
+I propose the following TypeScript declaration, not because it has any direct use,
+but purely to concretely define the data structure required:
+```ts
+declare class ThemeManifest {
+ // The name of the theme
+ name:        string;
+ // A more detail description for extra info about the theme
+ description: string;
+ // The Discord ID of the theme's author - only one allowed, prioritised over author
+ authorId?:   string;
+ // Who wrote the theme; not required to be just one person
+ author:      string;
+ // An SPDX identifier for the license of your theme content - https://spdx.org/licenses/
+ license:     string;
+ // The absolute URL of an image to go with your theme, screenshot or otherwise
+ media?:      URL;
```

Comment:
`media` should be an optional array, as being able to attach multiple screenshots is useful for showcasing a theme off

Reply:
i did consider this, the only reason it isnt is that, at the point of me writing that, we are/were using BD style manifests, which made this... difficult.

if we switch to JSON, i would expect a type of `string | string[]` to be more relevant

Reply:
at this point it seems to me that bd-style manifests are out of the question, so yes, i think that media will become an array.

(cc @lexisother @Cr3atable)